mirrord/layer/src/detour.rs

```diff
@@ -6,16 +13,52 @@ use std::{cell::RefCell, ops::Deref, os::unix::prelude::*, path::PathBuf};
 
 use crate::error::HookError;
 
-thread_local!(pub(crate) static DETOUR_BYPASS: RefCell<bool> = RefCell::new(false));
+thread_local!(
+    /// Holds the thread-local state for bypassing the layer's detour functions.
+    ///
+    /// ## Warning
+    ///
+    /// Do **NOT** use this directly, instead use `DetourGuard::new` if you need to
+    /// create a bypass inside a function (like we have in
+    /// [`TcpHandler::create_local_stream`](crate::tcp::TcpHandler::create_local_stream)).
```

Comment:
Maybe add a code example instead?

Reply:
Should I prevent access to this? Think we can restrict its visibility, and then leave this as a warning for people to not `pub(crate)` it in the future.

Reply:
Sounds good